matlab_simulation/04-sensor/scanner_bresenham/robot_motion.m

```diff
@@ -0,0 +1,108 @@
+function [X] = robot_motion( map, X, x0, T)
+
+% This function create a random robot position and a random robot motion
+% map -- real map
+% X -- robot states
+% x0 -- initial robot state
+% T -- time of simulation
+
+%% Randomly generate robot x & y velocities
+ 
+ x_vel = 1; % Initial x velocity
+ y_vel = 1; % Initial y velocity
+ theta_vel = 0.2; % Spinning rate
+ 
+ while (x_vel * y_vel == 0)
+    x_vel = 2*round(0.4*randn(1));
+    y_vel = 2*round(0.4*randn(1));
+ end
+
+ [M N] = size(map);
+
+ X(:,1) = x0; % Initial state of the robot
+ 
+ %%
+    mode = 5; % Robot initial moving mode
```

Comment:
Something happened with the spacing again here. Could use a bit of cleaning up, otherwise great.